wordle-web/scripts/validWordsUtils.ts

```diff
@@ -0,0 +1,168 @@
+    for (
+      let lettersIndex = 0;
+      lettersIndex < this.letters.length;
+      lettersIndex++
+    ) {
+      this.letters[lettersIndex].runningTotalDiscovered = 0;
+    }
+  }
+
+  validWords(): Array<string> {
+    this.setup();
+    // Eliminate words based off of previous guesses
+    let result = this.wordList.filter(wordListWord => {
+      let metCondition = this.filterWordList(wordListWord);
+      this.setRunningTotalForAllLettersToZero();
+      return metCondition;
+    });
+    // Eliminate words based off of current letters typed in to guess (not submitted)
+    result = result.filter(wordListWord => {
+      let lettersEntered = this.game.guesses[this.game.guessIndex].letters;
+      let lettersLength = lettersEntered.length;
+      let count = 0;
+      while (count < lettersLength && lettersEntered[count].char !== '') {
+        if (wordListWord[count].toUpperCase() !== lettersEntered[count].char) {
+          return false;
+        }
+        count++;
+      }
+      return true;
+    });
+    return result;
+  }
+
+  // Helper method to eliminate words based off of previous guesses
+  filterWordList(wordListWord: string): boolean {
+    let characters = wordListWord.toUpperCase().split('');
+    // Checks if it's valid for current index, based off of previous guess results
+    for (let index = 0; index < characters.length; index++) {
+      let character = characters[index];
+      let lettersIndex = character.charCodeAt(0) - 65;
+      let letterHelper = this.letters[lettersIndex];
+      letterHelper.runningTotalDiscovered++;
+      // Also checks if correct letters are found in the word
+      if (
+        (this.partialWord[index] !== undefined &&
+          this.partialWord[index] !== character) ||
+        !letterHelper.isValidForIndex[index]
+      ) {
+        return false;
+      }
+    }
+    // Makes sure there's at least however many letters already discovered (through misplaced and correct letters) in current word list word
+    for (
+      let lettersIndex = 0;
+      lettersIndex < this.letters.length;
+      lettersIndex++
+    ) {
+      let letter = this.letters[lettersIndex];
+      if (letter.runningTotalDiscovered < letter.numberDiscoveredInWord) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  // Set up valid for index, number found, number discovered based on current guesses
+  setup() {
+    for (
+      let guessesIndex = 0;
+      guessesIndex < this.game.guessIndex;
+      guessesIndex++
+    ) {
+      let guess = this.game.guesses[guessesIndex];
+      this.setRunningTotalForAllLettersToZero();
+      if (!guess.isFilled) {
+        return;
+      }
+      let lettersArray = guess.letters;
+      // Adjust letter helpers based on correct, misplaced, or wrong states
+      for (let index = 0; index < lettersArray.length; index++) {
+        let lettersIndex = lettersArray[index].char.charCodeAt(0) - 65;
+        if (lettersArray[index].state === LetterState.Correct) {

```

Comment:
```suggestion
        switch (lettersArray[index].state){
          case LetterState.Correct:
```
Breaking this `else if` chain up into a switch statement would help keep things a bit more organized and clear.